source/_integrations/flexmeasures.markdown

```diff
@@ -0,0 +1,171 @@
+  type: float
+soc_max:
+  description: The maximum state of charge that the flexible energy asset is allowed to reach.
+  required: true
+  type: float
+{% endconfiguration %}
+
+{% include integrations/config_flow.md %}
+
+## Schedule Sensor
+
+The Flexmeasures Schedule sensor shows the values of the power draw or supply at the start of each interval. It contains a startdatetime, unit of measurement, and a device class as well. The schedule is automatically shifted up to be in line with the sensor resolution. 
+
+```yaml
+schedule:
+  - start: '2023-09-04T17:45:00+02:00'
+    value: -0.5
+  - start: '2023-09-04T18:00:00+02:00'
+    value: -0.5
+start: '2023-09-04T17:45:00+02:00'
+unit_of_measurement: MWh
+device_class: energy
+friendly_name: FlexMeasures Schedule
+```
+
+## The scheduling service
+
+For a schedule to be calculated a `soc_at_start` is required. All other variables needed to calculate a new schedule were provided in the configuration. The following `yaml` file will trigger a schedule and update the sensor:
+
+```yaml
+service: flexmeasures.trigger_and_get_schedule
+data:
+  soc_at_start: "FLOAT_SOC_AT_START"
+```
+
+## Automate scheduling
+
+The intended usage of FlexMeasures is letting the schedules and Home Assistant optimize the flexible energy assets without user interaction. Automations can be used to trigger new schedules when new information is available that would impact the schedule. Some examples of events that should trigger the request of new schedules are when the battery/asset is connected, when new prices are available, and periodically to match the executed schedule to the calculated schedule. The user will need to provide a Home Assistant entity as a `soc_at_start` to be used for triggering a schedule. This is an example `yaml` for the action set in the automations:
+
+```yaml
+service: flexmeasures.trigger_and_get_schedule
+data:
+  soc_at_start: "\{\{ state_attr('SENSOR_TYPE.SENSOR', 'SENSOR_ATTRIBUTES') }}"
+```
+
+## Posting measurement data
+
+FlexMeasures needs data to calculate the optimal schedules for flexible assets. This data can be posted to the FlexMeasures instance using the `flexmeasures` integration. To post data the following `yaml` can be used:
+
+```yaml
+service: flexmeasures.post_measurements
+data:
+  sensor_id: SENSOR_TO_POST_DATA_TO_INT
+  start: "DATETIME_STRING_OF_FIRST_DATA_POINT"
+  duration: "DATETIME_DURATION_STRING_FOR_THE_COLLECTED_VALUES"
+  unit: "SENSOR_UNIT_OF_THE_VALUES"
+  prior: "DATETIME_STRING_AT_WHICH_THE_VALUES_WERE_RECORDED_OR_FORECASTED"
+```
+
+##S2 Protocol
+
+The S2 Protocol is a flexibility description protocol that allows a resource manager share their preferences with solutions that optimize energy assets (the Customer Energy Managers).
+
+This integration allows to have FlexMeasures exposed as a Customer Energy Manager just in HomeAssistant. That way, users can change the control type of a resource manager in HA interface.
+Changing the control type
```

Comment:
Besides my suggested rewrite in PR #2, this paragraph made me wonder whether our integration allows the S2 support to be toggled on or off (and if so, how), and what "changing the control type" is supposed to accomplish.

Reply:
maybe @victorgarcia98 knows but I can't explain it.